tests/bits/step-62_periodic.cc

```diff
@@ -0,0 +1,942 @@
+    , right_hand_side(dofs_per_cell)
+  {}
+
+
+
+  template <int dim>
+  ElasticWave<dim>::ElasticWave()
+    : mpi_communicator(MPI_COMM_WORLD)
+    , triangulation(mpi_communicator,
+                    typename Triangulation<dim>::MeshSmoothing(
+                      Triangulation<dim>::smoothing_on_refinement |
+                      Triangulation<dim>::smoothing_on_coarsening))
+    , fe_order(1)
+    , quadrature_formula(fe_order + 1)
+    , fe(FE_Q<dim>(fe_order), dim)
+    , dof_handler(triangulation)
+    , frequency(nb_frequency_points)
+    , periodicity_offset([](const double dimension_x) {
+      Tensor<1, dim> lambda_periodicity_offset;
+      lambda_periodicity_offset[0] = dimension_x;
+      lambda_periodicity_offset[1] = 0;
+      if (dim == 3)
+        {
+          lambda_periodicity_offset[2] = 0;
+        }
+      return lambda_periodicity_offset;
+    }(dimension_x))
+  {}
+
+
+
+  template <int dim>
+  void
+  ElasticWave<dim>::setup_system()
+  {
+    dof_handler.distribute_dofs(fe);
+
+    locally_owned_dofs = dof_handler.locally_owned_dofs();
+    DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant_dofs);
+
+    locally_relevant_solution.reinit(locally_owned_dofs,
+                                     locally_relevant_dofs,
+                                     mpi_communicator);
+
+    system_rhs.reinit(locally_owned_dofs, mpi_communicator);
+
+    constraints.clear();
+    constraints.reinit(locally_relevant_dofs);
+    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+
+    const std::complex<double> periodicity_factor =
+      std::exp(-std::complex<double>(0, 1) * wavenumber * dimension_x);
+    std::vector<
+      GridTools::PeriodicFacePair<typename DoFHandler<dim>::cell_iterator>>
+      matched_pairs;
+    GridTools::collect_periodic_faces(dof_handler,
+                                      left_boundary,
+                                      right_boundary,
+                                      periodicity_direction,
+                                      matched_pairs,
+                                      periodicity_offset);
+    //    DoFTools::make_periodicity_constraints<DoFHandler<dim>>(
+    //     matched_pairs,
+    //     constraints,
+    //     ComponentMask(),
+    //     std::vector<unsigned int>(),
+    //     periodicity_factor);
+
+    // Workaround, instead of using make_periodicity_constraints. We set
+    // the constraints by hand.
+    unsigned int  dof_left;
+    unsigned int  dof_right;
+    std::ifstream file;
+    file.open("/tmp/dofs");
+    for (unsigned int idx = 0; idx < 340; ++idx)
+      {
+        file >> dof_left;  // reads in the double value
+        file >> dof_right; // reads in the double value
+        std::cout << dof_left << " i " << dof_right
+                  << std::endl; // for test purposes
+        constraints.add_line(dof_left);
+        constraints.add_entry(dof_left, dof_right, periodicity_factor);
```

Comment:
This is a workaround. If I apply the constraints by hand. The test works. As you can see I commented `make_periodicity_constraints()`